builder/build/main.rs

```diff
@@ -0,0 +1,207 @@
+        fs::create_dir_all(Path::new(self.target_lib.as_ref()))
+            .expect("Failed to create include directory");
+        fs::create_dir_all(Path::new(self.target_bin.as_ref()))
+            .expect("Failed to create include directory");
+        fs::create_dir_all(Path::new(self.target_pkconfig.as_ref()))
+            .expect("Failed to create include directory");
+    }
+
+    fn deduplicate_headers(&self) {
+        let datadog_inc_dir = Path::new(self.source_inc.as_ref());
+
+        let mut headers: Vec<String> = Vec::new();
+        let inc_files = fs::read_dir(datadog_inc_dir).unwrap();
+        for file in inc_files.flatten() {
+            let name = file.file_name().into_string().unwrap();
+            if name.ends_with(".h") && !name.eq("common.h") && !name.eq("blazesym.h") {
+                headers.push(file.path().to_string_lossy().to_string());
+            }
+        }
+
+        let base_header = self.source_inc.to_string() + "/" + self.main_header.as_ref();
+        dedup_headers(&base_header, &headers);
+    }
+
+    // TODO: maybe do this in module's build.rs
+    fn sanitize_libraries(&self) {
+        let datadog_lib_dir = Path::new(self.source_lib.as_ref());
+
+        let libs = fs::read_dir(datadog_lib_dir).unwrap();
+        for lib in libs.flatten() {
+            let name = lib.file_name().into_string().unwrap();
+            if name.ends_with(".so") {
+                arch::fix_rpath(lib.path().to_str().unwrap());
+            }
+        }
+    }
+
+    fn add_cmake(&self) {
+        let libs = arch::NATIVE_LIBS.to_owned();
+        let output = Command::new("sed")
+            .arg("s/@Datadog_LIBRARIES@/".to_string() + libs.trim() + "/g")
+            .arg("../cmake/DatadogConfig.cmake.in")
+            .output()
+            .expect("Failed to modify cmake");
+
+        let cmake_path: PathBuf = [&self.target_dir, "DatadogConfig.cmake"].iter().collect();
+        fs::write(cmake_path, output.stdout).expect("writing cmake file failed");
+    }
+
+    fn build(&self) -> Result<()> {
+        for module in &self.modules {
+            module.install()?;
+        }
+        Ok(())
+    }
+
+    fn pack(&self) -> Result<()> {
+        let tarname = "libdatadog".to_string() + "_v" + &self.version + ".tar";
+        let path: PathBuf = [self.target_dir.as_ref(), &tarname].iter().collect();
+        let artifact = fs::File::create(path).expect("Failed to create tarfile");
+        let mut ar = tar::Builder::new(artifact);
+        ar.append_dir_all("lib", self.target_lib.as_ref())?;
+        ar.append_dir("bin", self.target_bin.as_ref())?;
+        ar.append_dir_all("include", self.target_include.as_ref())?;
+
+        Ok(ar.finish().expect("Failed to write the tarfile"))
+    }
+}
+
+fn main() {
+    let mut path = env::var("OUT_DIR").unwrap();
+    if let Ok(libdd_path) = env::var("LIBDD_OUTPUT_FOLDER") {
+        path = libdd_path;
+    }
+
+    let profile = env::var("PROFILE").unwrap();
```

Comment:
I had a determine path function in the build-common parts to try and have a common way of defining where we should output files.
What do you think of the logic that is available there ?